api/api-stage-operations-v1/src/main/java/com/thoughtworks/go/apiv1/stageoperations/StageOperationsControllerV1.java

```diff
@@ -73,4 +137,38 @@ public String run(Request req, Response res) throws IOException {
         scheduleService.rerunStage(pipelineName, pipelineCounter, stageName, result);
         return renderHTTPOperationResult(result, req, res);
     }
+
+    private Optional<Stage> getStageFromRequestParam(Request request, HttpOperationResult operationResult) {
+        String pipelineName = request.params("pipeline_name");
+        String pipelineCounter = request.params("pipeline_counter");
+        String stageName = request.params("stage_name");
+        String stageCounter = request.params("stage_counter");
+
+        Stage stage = stageService.findStageWithIdentifier(pipelineName,
+                Integer.parseInt(pipelineCounter),
+                stageName,
+                stageCounter,
+                currentUsername().getUsername().toString(),
+                operationResult);
+
+        if (!operationResult.isSuccess()) {
+            return Optional.empty();
+        }
+
+        if (stage == null || stage instanceof NullStage) {
+            String message = String.format("Stage '%s' not found", stageName);
```

Comment:
I think this message should change, we could get a NullStage even for a wrong stage counter.